src/views/ChartPage.vue

```diff
@@ -0,0 +1,463 @@
+        if (nb > 3) {
+          listColor.splice(2, 0, '#27ae60')
+          listColor.push('#fb19cb')
+          if (nb > 5) {
+            listColor.splice(2, 0, '#8e44ad')
+            if (nb > 6) {
+              listColor.splice(2, 0, '#fba619')
+              if (nb > 7) {
+                listColor.splice(6, 0, '#152c8c')
+                listColor.push('#935116', '#fdfefe', '#17202a')
+              }
+            }
+          }
+        }
+      }
+      return listColor
+    },
+    drawPieChart () {
+      const labelList = this.getNames(this.masterCategoryId)
+      const dataList = this.getDatas(this.typeInformationPie, this.masterCategoryId)
+      const colorList = this.getColor(false, false, false, dataList.length)
+      this.pieChartData.labels = labelList
+      this.pieChartData.datasets[0].data = dataList
+      this.pieChartData.datasets[0].backgroundColor = colorList
+    },
+    drawBarChart () {
+      this.chartData.datasets = []
+      if (this.typeInformationBar.indexOf('available') !== -1) {
+        const datasetsAvailable = {
+          label: 'Disponible',
+          backgroundColor: this.getColor(false, false, true, 0),
+          data: this.getDatas('available', this.masterCategoryId)
+        }
+        this.chartData.datasets.splice(0, 0, datasetsAvailable)
+      }
+      if (this.typeInformationBar.indexOf('spent') !== -1) {
+        const datasetsSpent = {
+          label: 'Dépensé',
+          backgroundColor: this.getColor(false, true, false, 0),
+          data: this.getDatas('spent', this.masterCategoryId)
+        }
+        this.chartData.datasets.splice(0, 0, datasetsSpent)
+      }
+      if (this.typeInformationBar.indexOf('allocated') !== -1) {
+        const datasetsAllocation = {
+          label: 'Allocation',
+          backgroundColor: this.getColor(true, false, false, 0),
+          data: this.getDatas('allocated', this.masterCategoryId)
+        }
+        this.chartData.datasets.splice(0, 0, datasetsAllocation)
+      }
+      this.chartData.labels = this.getNames(this.masterCategoryId)
+    },
+    changeGraph () {
+      switch (this.currentGraph) {
+        case 'pie':
+          this.currentGraph = 'bar'
+          break
+        default:
+          this.currentGraph = 'pie'
+      }
+    },
+    getMonthAsDate (monthAsInt: number): Date {
+      return Time.getMonthAsDate(monthAsInt)
+    },
+    async goToNextMonth () {
+      this.budgetMonth = Time.getNextMonth(this.budgetMonth)
+      await this.getBudgetData()
+      this.recalculate()
+    },
+    async goToLastMonth () {
+      this.budgetMonth = Time.getLastMonth(this.budgetMonth)
+      await this.getBudgetData()
+      this.recalculate()
+    },
+    recalculate () {
+      this.drawPieChart()
+      this.drawBarChart()
+    },
+    switchTypeInformationPie (newType: string) {
+      switch (newType) {
+        case 'allocated':
+          this.typeInformationPie = 'allocated'
```

Comment:
why not set typeInformationPie as newType without the case ?

Reply:
done